module/BingoArea.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file BingoArea.cpp
+ * @brief ビンゴエリアの情報を保持するクラス
+ * @author Hisataka-Hagiyama,uchyam
+ */
+
+#include "BingoArea.h"
+BingoArea::BingoArea()
+{
+  //交点サークルにおけるサークルの色の初期値
+  std::array<std::array<COLOR, 4>, 4> crossCircleColor
+      = { { { COLOR::RED, COLOR::RED, COLOR::BLUE, COLOR::BLUE },
+            { COLOR::RED, COLOR::RED, COLOR::BLUE, COLOR::BLUE },
+            { COLOR::YELLOW, COLOR::YELLOW, COLOR::GREEN, COLOR::GREEN },
+            { COLOR::YELLOW, COLOR::YELLOW, COLOR::GREEN, COLOR::GREEN } } };
+
+  //黒線の数の初期値
+  std::array<std::array<int, 4>, 4> initEdgeNumber
+      = { { { 2, 3, 3, 2 }, { 3, 4, 4, 3 }, { 3, 4, 4, 3 }, { 2, 3, 3, 2 } } };
+
+  //ブロックサークルにおけるサークルの色の初期値
+  std::array<COLOR, 8> blockCircleColor{ COLOR::YELLOW, COLOR::GREEN, COLOR::RED, COLOR::BLUE,
+                                         COLOR::YELLOW, COLOR::GREEN, COLOR::RED, COLOR::BLUE };
+
+  //交点サークルの座標、サークルの色、黒線の本数をセット
+  int index = 0, index2 = 0, index3 = 0;
+  for(int y = 0; y < 7; y += 2) {
+    for(int x = 0; x < 7; x += 2) {
+      crossCircle[index].setCoordinate(x, y);
+      crossCircle[index].setCircleColor(crossCircleColor[index2][index3]);
+      crossCircle[index].setEdgeNumber(initEdgeNumber[index2][index3]);
+      index3++;
+      index++;
+    }
+    index3 = 0;
+    index2++;
+  }
+
+  //ブロックサークルの座標、サークルの色をセット
+  int index4 = 0, index5 = 0;
+  for(int y = 1; y < 6; y += 2) {
+    for(int x = 1; x < 6; x += 2) {
+      if(x == 3 && y == 3) {
+        centerMark.setCoordinate(x, y);
+        centerMark.setCircleColor(COLOR::BLACK);
+        index5++;
+        continue;
+      }
+      blockCircle[index4].setCoordinate(x, y);
+      blockCircle[index4].setCircleColor(blockCircleColor[index4]);
+      blockCircle[index4].setCircleId(static_cast<CIRCLE_ID>(index5));
+      index4++;
+      index5++;
+    }
+  }
+}

```

Comment:
ここの処理が良く分かんないんだけど，↓みたいな処理ではダメ?
あと，CIRCLE_IDって0~7ってNode.hで書いてあるけど，この処理だと8まで行かない？

他でblockCircleを使ってるところ見てみても，blockCicleのインデックスをそのままIDとして扱えるような気がするんだけど，そもそもCIRCLE_IDって何のためにあるの？


```suggestion
  int index4 = 0;
  for(int y = 0; y < 3; y ++) {
    for(int x = 0; x < 3; x ++) {
      if(x == 1 && y == 1) {
        centerMark.setCoordinate(x * 2 + 1, y * 2 + 1);
        centerMark.setCircleColor(COLOR::BLACK);
      }else {
        blockCircle[index4].setCoordinate(x * 2 + 1, y * 2 + 1);
        blockCircle[index4].setCircleColor(blockCircleColor[index4]);
        blockCircle[index4].setCircleId(static_cast<CIRCLE_ID>(index4));
      }
    }
  }

```

Reply:
こっちも`index4++`してなかった

Reply:
int型よりenum classの方が-1〜7で制限されるから安全。という理由でCIRCLE_IDを作りました